src/openforms/formio/rendering/utils.py

```diff
@@ -0,0 +1,53 @@
+from typing import TYPE_CHECKING
+
+from glom import Assign, Path, glom
+
+from openforms.submissions.rendering import Renderer, RenderModes
+from openforms.submissions.rendering.nodes import SubmissionStepNode
+from openforms.typing import JSONObject
+
+from .default import ColumnsNode, EditGridGroupNode, EditGridNode, FieldSetNode
+from .nodes import ComponentNode
+
+if TYPE_CHECKING:
+    from openforms.submissions.models import Submission
+
+
+def render_nested_json(submission: "Submission") -> JSONObject:
+    """Render submission as JSON with nesting
+
+    The data is nested within each submission step (using the form definition slug as key).
+    The data is nested for fieldset components and for column components.
```

Comment:
maybe add a remark that this is different from how formio itself treats these? This is more operating on the level of Formio's data container component